main.go

```diff
@@ -215,8 +234,16 @@ func main() {
 			testModel.EnvironmentMatrix.IosDeviceList.IosDevices = append(testModel.EnvironmentMatrix.IosDeviceList.IosDevices, &newDevice)
 		}
 
+		timeout := configs.TestTimeout
```

Comment:
please move this validation part to the input validations, as if this input contains something wrong then we should fail ASAP, and also it makes more sense there maybe